skbio/alignment/_path.py

```diff
@@ -0,0 +1,308 @@
+
+        # TODO: An additional parameter `starts` should record the starting positions
+        # of the alignment in each sequence (shape: (n_seqs,)). This is important
+        # especially for local alignments (e.g., search for a short read in a genome
+        # sequence). If it is provided, `n_seqs` is no longer necessary.
+        # self.starts = starts
+
+        # TODO: Needs to think about whether reverse complemented (Boolean array of
+        # (n_seqs,)) should be included as a parameter. It is only relevant for
+        # nucleotide sequences.
+
+    def __str__(self):
+        """Return string representation of this AlignPath."""
+        # Not sure if this makes sense for this class, but it is needed for all
+        # SkbioObjects.
+        return self.__repr__()
+
+    def __repr__(self):
+        """Return summary of the alignment path."""
+        return (
+            f"<{self.__class__.__name__}, shape: {self.shape}, lengths: "
+            f"{self.lengths}, states: {self.states}"
+        )
+
+    def subset(self):
+        """Select subset of sequences from AlignPath.
+
+        Better to have ability to index an AlignPath for particular sequences,
+        then compress these sequences using something like _fix_arrays"""
+        pass
+
+    def to_bits(self):
+        """Unpack states into an array of bits."""
+        return np.unpackbits(
+            np.atleast_2d(self.states), axis=0, count=self.shape[0], bitorder="little"
+        )
+
+    @classonlymethod
+    def from_bits(cls, bits):
+        """Create an alignment path from a bit array (0 - char, 1 - gap)."""
+        # Pack bits into integers.
+        ints = np.packbits(bits, axis=0, bitorder="little")
+
+        # If there are 8 or less sequences, squeeze the 2D array into 1D.
+        # This is an optional optimization especially for pairwise alignment.
+        if ints.shape[0] == 1:
+            ints = np.squeeze(ints, axis=0)
+
+            # Get indices where segments start. Equivalent to but faster than:
+            # idx = np.where(np.diff(ints, prepend=np.nan))[0]
+            idx = np.append(0, np.where(ints[:-1] != ints[1:])[0] + 1)
+
+            # Get lengths of segments. Equivalent to but faster than:
+            # lens = np.diff(idx, append=ints.size)
+            lens = np.append(idx[1:] - idx[:-1], ints.size - idx[-1])
+
+            # Keep indices of segment starts.
+            ints = ints[idx]
+
+        # This is the 2D equivalent of the above code.
+        else:
+            idx = np.append(
+                0, np.where((ints[:, :-1] != ints[:, 1:]).any(axis=0))[0] + 1
+            )
+            lens = np.append(idx[1:] - idx[:-1], ints.shape[1] - idx[-1])
+            ints = ints[:, idx]
+
+        # return per-segment lengths and bits
+        return cls(lens, ints, np.zeros(bits.shape[0], dtype=int))
+
+    @classonlymethod
+    def from_tabular(cls, msa):
+        """Create an alignment path from a `TabularMSA` object."""
```

Comment:
Is it worthwhile to have a `to_tabular` version method? It maybe ideal for easier manipulation / visualization